### PR TITLE
Fix Memcache/MemcachePool get method signature

### PR DESCRIPTION
**Pull request by dravnic**
https://www.php.net/manual/en/memcache.get.php

### CODE REVIEW

File: resources/functionMap.php [hunk 1]
```diff
@@ -6414,7 +6414,7 @@
 'Memcache::decrement' => ['int', 'key'=>'string', 'value='=>'int'],
 'Memcache::delete' => ['bool', 'key'=>'string', 'timeout='=>'int'],
 'Memcache::flush' => ['bool'],
-'Memcache::get' => ['string|array|false', 'key'=>'string', '&flags='=>'array', '&keys='=>'array'],
+'Memcache::get' => ['string|array|false', 'key'=>'string|array', '&flags='=>'int|array'],
```
Comment (staabm) [inline resources/functionMap.php]: Could we get more precise, e.g. string[] ?

Comment (dravnic) [reply resources/functionMap.php]: @staabm Good idea. Just added commit with string[] for key and int[] for flags

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Can you please add these as separate variants? Because they are separate variants at https://www.php.net/manual/en/memcache.get.php

What I mean by that - look at the top of the file and see the function `abs()`:

```
'abs' => ['0|positive-int', 'number'=>'int'],
'abs\'1' => ['float', 'number'=>'float'],
'abs\'2' => ['float|0|positive-int', 'number'=>'string'],
```

You can also test that in NodeScopeResolverTest - passing string should return string, passing array should return array.

**Comment by dravnic** [comment]
@ondrejmirtes Sure, I didn't know it was possible to define variants. makes more sense.

**Comment by dravnic** [comment]
Using variants in the latest commit. 

Also fixed return value. Memcache::get deserializes so return value should be mixed i.e. mixed[]|false

**Comment by ondrejmirtes** [comment]
You can also test that in NodeScopeResolverTest - passing string should return string, passing array should return array.

**Comment by dravnic** [comment]
It seems that PHP documentation isn't correct regarding `Memcache::get` return type. It doesn't return `string|array`, but `mixed|mixed[]` type depending on value. Below is a simple script that checks this. 

How to test this in NodeScopeResolverTest? Is there a stub/mock of Memcache class that I could use? I didn't find any.

```php
<?php
declare(strict_types=1);

$memcache = new Memcache();
$memcache->connect('localhost');

$memcache->set('key-number', 123);
$memcache->set('key-string', "abc");
$memcache->set('key-array', array(1,2,3));
$memcache->set('key-object', (object) array(1,2,3));

echo "\n---- Memcache::get (single) ----\n";
echo "key-number type -> " . gettype($memcache->get('key-number')) . "\n";
echo "key-string type -> " . gettype($memcache->get('key-string')) . "\n";
echo "key-array type -> " . gettype($memcache->get('key-array')) . "\n";
echo "key-object type -> " . gettype($memcache->get('key-object')) . "\n";

echo "\n---- Memcache::get (multi)  ----\n";
$multi = $memcache->get(array('key-number', 'key-string', 'key-array', 'key-object'));
foreach ($multi as $key => $value) {
    echo "$key type -> " . gettype($value) . "\n";
}

```

and here is the output of running it (php 8.1):

```
---- Memcache::get (single) ----
key-number type -> integer
key-string type -> string
key-array type -> array
key-object type -> object

---- Memcache::get (multi)  ----
key-number type -> integer
key-string type -> string
key-array type -> array
key-object type -> object

```

**Comment by ondrejmirtes** [comment]
Here's the documentation about testing: https://phpstan.org/developing-extensions/testing#type-inference

You can also check out all the files linked in NodeScopeResolverTest to get the idea how it works.

**Comment by dravnic** [comment]
@ondrejmirtes just added tests to the NodeScopeResolverTest. hope it's ok now.

**Comment by ondrejmirtes** [comment]
@dravnic 1) Please base it on top of 1.10.x.
2) Rebase it so that it contains only your own commits and no merge commits.

**Pull request closed without merging**